apps/tools/blocks-crawler/src/blocks.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use mina_rs_base::types::ExternalTransition;
+use mina_rs_base::JsonSerializationType;
+use serde::{Deserialize, Serialize};
+use serde_json::Error;
+use std::path::Path;
+
+// To read query response from graphql api (https://graphql.minaexplorer.com/)
+#[derive(Debug, Serialize, Deserialize)]
+pub struct QueryResponse {
+    pub data: Blocks,
+}
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Blocks {
+    pub blocks: Vec<BlockInfo>,
+}
+#[derive(Debug, Serialize, Deserialize)]
+pub struct BlockInfo {
+    #[serde(rename = "blockHeight")]
+    pub block_height: usize,
+    #[serde(rename = "stateHash")]
+    pub state_hash: String,
+}
+
+impl std::fmt::Display for BlockInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}-{}", self.block_height, self.state_hash)
+    }
+}
+
+pub async fn check_block(network: &str, block: BlockInfo) {
+    // Check for previously failed block and delete it to check again
+    let local_failed_block = format!("output/failed/{}-{}.json", network, block);
+    if Path::new(&local_failed_block).exists() {
+        match std::fs::remove_file(&local_failed_block) {
+            Ok(_) => info!("Deleted {}", local_failed_block),
+            Err(err) => error!("Failed to delete {}: {}", local_failed_block, err),
+        }
+    }
+    // Check for previously succeeded block
+    let local_succeeded_block = format!("output/succeeded/{}-{}.json", network, block);
+    if !Path::new(&local_succeeded_block).exists() {
+        info!("Checking: {}", block);
+        let block_url = format!(
+            "https://storage.googleapis.com/mina_network_block_data/{}-{}.json",
+            network, block
+        );
+        // Retrieve block json from mainnet json dump
+        match retrieve_block_json(&block_url).await {
+            Ok(block_json) => match parse_block_json(&block_json) {
+                Ok(_et) => {
+                    write_json("succeeded", "mainnet", &block, &block_json);
```

Comment:
can use the `network` argument instead of the literal `"mainnet"`

Reply:
done